packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
I admit that I don't completely understand what is being suggested/discussed here. But my naive take would be that the runfunction should not take an error argument. We should just be able to handle that by rejecting the Promise returned. Also I am not sure I understand why does user need to call commit. Even in the non promise case, i dont undertstand the need to pass error to run function.